testplan/runnable/interactive/http.py

```diff
@@ -952,6 +952,10 @@ def _extract_entries(entry: Dict) -> Dict:
     """
     entries = {}
 
+    # FIXME:
+    # with ReportCategories.SYNTHESIZED being added, following line is no longer
+    # able to tell testcase entries from assertion entries; while filtering has
+    # been properly done in frontend code, it should just work in production
```

Comment:
It seems frontend sends testcase report with entries (and garbage data) when filter is applied.

So you may go recursive into the garbage data when entry is a synthesized testcase level report.